File: ui/src/main/java/com/peakmain/ui/utils/database/DaoSupportFactory.java
package com.peakmain.ui.utils.database;

import android.database.sqlite.SQLiteDatabase;
import android.os.Environment;
import android.util.Log;

import com.peakmain.ui.utils.LogUtils;

import java.io.File;

public class DaoSupportFactory {
    private static volatile DaoSupportFactory mFactory;
    private File mFile;

    public static DaoSupportFactory getInstance() {
        if (mFactory == null) {
            synchronized (DaoSupportFactory.class) {
                if (mFactory == null) {
                    mFactory = new DaoSupportFactory();
                }
            }
        }
        return mFactory;
    }

    private SQLiteDatabase mSqliteDatabase;

    private DaoSupportFactory() {
        if (mFile == null)
            mFile = new File(Environment.getExternalStorageDirectory().getAbsolutePath() + File.separator
                    + "BaseUI" + File.separator + "database");
        LogUtils.e(mFile.getAbsolutePath());
        if (!mFile.exists()) {
            mFile.mkdirs();
        }
        File dbFile = new File(mFile, "baseUI.db");
        mSqliteDatabase = SQLiteDatabase.openOrCreateDatabase(dbFile, null);
    }

    public <T> IDaoSupport<T> getDao(Class<T> clazz) {
        IDaoSupport<T> daoSupport = new DaoSupport();
        daoSupport.init(mSqliteDatabase, clazz);
        return daoSupport;
    }

    /**
     * 设置保存的文件目录
     *
     * @param file 默认是/BaseUI/database
     */
    public void setFileDir(File file) {
        this.mFile = file;
    }

    /**
     * 设置保存的文件目录
     *
     * @param pathName 保存的目录，默认是/BaseUI/database
     */
    public void setFileDir(String pathName) {
        this.mFile = new File(pathName);
    }
}

File: ui/src/main/java/com/peakmain/ui/utils/database/DaoSupportFactory.kt
package com.peakmain.ui.utils.database

import android.database.sqlite.SQLiteDatabase
import android.os.Environment
import com.peakmain.ui.utils.LogUtils
import java.io.File

class DaoSupportFactory private constructor() {
    private var mFile: File? = null
    private val mSqliteDatabase: SQLiteDatabase
    fun <T> getDao(clazz: Class<T?>?): IDaoSupport<T?> {
        val daoSupport: IDaoSupport<T?> = DaoSupport()
        daoSupport.init(mSqliteDatabase, clazz)
        return daoSupport
    }

    /**
     * 设置保存的文件目录
     *
     * @param file 默认是/BaseUI/database
     */
    fun setFileDir(file: File?) {
        mFile = file
    }

    /**
     * 设置保存的文件目录
     *
     * @param pathName 保存的目录，默认是/BaseUI/database
     */
    fun setFileDir(pathName: String?) {
        mFile = File(pathName)
    }

    companion object {
        @Volatile
        private var mFactory: DaoSupportFactory? = null
        @JvmStatic
        val instance: DaoSupportFactory?
            get() {
                if (mFactory == null) {
                    synchronized(DaoSupportFactory::class.java) {
                        if (mFactory == null) {
                            mFactory = DaoSupportFactory()
                        }
                    }
                }
                return mFactory
            }
    }

    init {
        if (mFile == null) mFile = File(Environment.getExternalStorageDirectory().absolutePath + File.separator
                + "BaseUI" + File.separator + "database")
        LogUtils.e(mFile!!.absolutePath)
        if (!mFile!!.exists()) {
            mFile!!.mkdirs()
        }
        val dbFile = File(mFile, "baseUI.db")
        mSqliteDatabase = SQLiteDatabase.openOrCreateDatabase(dbFile, null)
    }
}